src/pages/en/guides/migrate-to-astro/from-nextjs.mdx

```diff
@@ -2,72 +2,924 @@
+
+- The `pages` directory
+
+- [The `/app` directory (in beta)](https://beta.nextjs.org/docs/routing/pages-and-layouts#layouts)
+
+Each Astro page explicitly requires `<html>`, `<head>`, and `<body>` tags to be present, so it is common to reuse a layout file across pages. Astro uses a [`<slot />`](/en/core-concepts/astro-components/#slots) for page content, with no import statement required. Note the standard HTML templating, and direct access to `<head>`:  
+
+```astro title="src/layouts/Layout.astro"
+---
+---
+<html lang="en">
+	<head>
+		<meta charset="utf-8" />
+		<link rel="icon" type="image/svg+xml" href="/favicon.svg" />
+		<meta name="viewport" content="width=device-width" />
+		<meta name="generator" content={Astro.generator} />
+		<title>Astro</title>
+	</head>
+	<body>
+    <!-- Wrap the slot element with your existing layout templating -->
+		<slot />
+	</body>
+</html>
+```
+
+#### Migrating from Next.js' `pages` directory
+
+Your Next project may have a `pages/_document.jsx` file that imports React components to customize your app's `<head>`:
+
+```jsx title="pages/_document.jsx"
+import Document, { Html, Head, Main, NextScript } from "next/document";
+
+export default class MyDocument extends Document {
+  render() {
+    return (
+      <Html lang="en">
+        <Head>
+          <link rel="icon" href="/favicon.ico" />
+        </Head>
+        <body>
+          <Main />
+          <NextScript />
+        </body>
+      </Html>
+    );
+  }
+}
+```
+
+1. Make a new Astro layout file using only the returned JSX.
+
+2. Replace any React components with `<html>`, `<head>`, `<slot>`, and other HTML standard tags.
+
+
+```astro title="src/layouts/Document.astro"
+<html lang="en">
+	<head>
+	    <link rel="icon" href="/favicon.ico" />
+	</head>
+	<body>
+		<slot/>
+	</body>
+</html>
+```
+
+#### Migrating from Next.js' `/app` directory
+
+Next.js' `app/` directory layout files are created with two files: a `layout.jsx` file to customize the `<html>` and `<body>` contents, and a `body.jsx` file to customize the `<head>` element contents.
+
+```jsx title="app/layout.jsx"
+export default function Layout({ children }) {
+  return (
+    <html lang="en">
+      <body>{children}</body>
+    </html>
+  );
+}
+```
+
+```jsx title="app/head.jsx"
+export default function Head() {
+  return (
+    <>
+      <title>My Page</title>
+    </>
+  );
+}
+```
+
+1. Make a new Astro layout file using only the returned JSX.
+
+2. Replace both these files with a single Astro layout file that contains a page shell (`<html>`, `<head>`, and `<body>` tags) and a `<slot/>` instead of React's `{children}` prop:
+
+```astro title="src/layouts/Layout.astro"
+<html lang="en">
+	<head>
+	    <title>My Page</title>
+	</head>
+	<body>
+		<slot/>
+	</body>
+</html>
+```
+
+### Migrating Pages and Posts
+
+In Next.js, your posts either live in `/pages` or `/app/routeName/page.jsx`.
+
+In Astro, **all your page content must live within `src/`**.
```

Comment:
```suggestion
In Astro, **all your page content must live within `src/`**, in either `src/pages` or `src/content`.
```
This makes it clear that you're not just dropping them in the root of the source, and also primes you to expect two ways to store Astro pages.